### PR TITLE
tcpproxy.proto add protocol field to support multi-protocol proxy

### PR DESCRIPTION
**Pull request by Poorunga**
1. tcpproxy.proto is renamed to proxy.proto, and add the protocol field, then regenerate proxy.pb.go
2. add establishProxyConn to support multi-protocol proxy

### CODE REVIEW

File: agent/pkg/tunnel/proxy/proxy.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	var err error
+	var proxyConn net.Conn
+
+	switch msg.GetProtocol() {
+	case "tcp":
+		tcpAddr := &net.TCPAddr{
+			IP:   net.ParseIP(msg.GetIp()),
+			Port: int(msg.GetPort()),
+		}
+		for i := 0; i < MaxRetryTime; i++ {
```
Comment (khalid-jobs) [inline agent/pkg/tunnel/proxy/proxy.go:101]: sleep some seconds then retry

File: agent/pkg/tunnel/proxy/proxy.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+	"github.com/kubeedge/edgemesh/agent/pkg/tunnel/proxy/pb"
+	"github.com/kubeedge/edgemesh/common/constants"
+)
+
+const (
+	MaxReadSize  = 4096
+	MaxRetryTime = 3
+)
+
+var ProxyProtocol protocol.ID = "/libp2p/edgemesh/1.0.0"
```
Comment (khalid-jobs) [inline agent/pkg/tunnel/proxy/proxy.go:28]: the protocol name should indicate what the protocol does， such as `/libp2p/circuit/relay/0.1.0` for circuit relay

File: agent/pkg/tunnel/proxy/proxy.go (hunk 1 above)
Comment (Poorunga) [reply agent/pkg/tunnel/proxy/proxy.go:101]: fix

File: agent/pkg/tunnel/proxy/proxy.go (hunk 2 above)
Comment (Poorunga) [reply agent/pkg/tunnel/proxy/proxy.go]: fix

**Pull request merged**